nixos/modules/services/misc/godns.nix

```diff
@@ -0,0 +1,182 @@
+        default = null;
+        description = "Password of the DNS provider.";
+      };
+
+      passwordFile = mkOption {
+        type = types.nullOr types.str;
+        default = null;
+        description = "Path to file containing password of the DNS provider.";
+      };
+
+      loginToken = mkOption {
+        type = types.nullOr types.str;
+        default = null;
+        description = "API token of the DNS provider.";
+      };
+
+      loginTokenFile = mkOption {
+        type = types.nullOr types.str;
+        default = null;
+        description = "Path to file containing API token of the DNS provider.";
+      };
+
+      domains = mkOption {
+        type = with types; listOf (submodule { options = {
+          domainName = mkOption {
+            type = str;
+            description = "Name of the domain to update.";
+          };
+          subDomains = mkOption {
+            type = listOf str;
+            description = "List of subdomains to update.";
+          };
+        }; });
+        default = [];
+        description = "Domains list, with your sub domains.";
+      };
+
+      ipv4Url = mkOption {
+        type = types.str;
+        default = "https://ip4.seeip.org";
+        description = "A URL for fetching one's public IPv4 address.";
+      };
+
+      ipv6Url = mkOption {
+        type = types.str;
+        default = "https://ip6.seeip.org";
+        description = "A URL for fetching one's public IPv6 address.";
+      };
+
+      ipType = mkOption {
+        type = types.enum [ "IPv4" "IPv6" ];
+        default = "IPv4";
+        description = "Switch deciding if IPv4 or IPv6 should be used.";
+      };
+
+      interval = mkOption {
+        type = types.int;
+        default = 300;
+        description = "How often (in seconds) the public IP should be updated.";
+      };
+
+      resolver = mkOption {
+        type = types.str;
+        default = "8.8.8.8";
+        description = "Address of a public DNS server to use.";
+      };
+
+      extraConf = mkOption {
+        type = types.attrs;
+        default = {};
+        description = "GoDNS extra configuration";
+      };
+
+      user = mkOption {
+        type = types.str;
+        default = "godns";
+        description = "User under which godns runs.";
+      };
+
+      group = mkOption {
+        type = types.str;
+        default = "godns";
+        description = "Group under which godns runs.";
+      };
+
+      dataDir = mkOption {
+        type = types.path;
+        default = "/var/lib/godns";
+        description = "GoDNS data directory.";
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    systemd.tmpfiles.rules = [
+      "d '${cfg.dataDir}' 0700 ${cfg.user} ${cfg.group} - -"
+    ];
+
+    systemd.services.godns = {
+      description = "GoDNS";
+      after = [ "network.target" ];
+      wantedBy = [ "multi-user.target" ];
+
+      serviceConfig = mkForce {
+        WorkingDirectory = cfg.dataDir;
```

Comment:
I'm not sure. Seems godns won't create or read files in its working directory? Is this `WorkingDirectory` necessary?